courseworks/week2/ZhigaoXIE/Q1.md

```diff
@@ -0,0 +1,10 @@
+| Type                      | is concrete | is primitive | is abstract | is bits type | is mutable |
+|---------------------------|-------------|--------------|-------------|--------------|-------------|
+| ComplexF64                | true        | false        | false       | true         | true        |
+| Complex{AbstractFloat}    | false       | false        | true        | false        | true        |
+| Complex{<:AbstractFloat}  | false       | false        | true        | false        | true        |
+| AbstractFloat             | false       | false        | true        | false        | false       |
+| Union{Float64, ComplexF64}| false       | false        | false       | true         | false       |
+| Int32                     | true        | true         | false       | true         | true        |
+| Matrix{Float32}           | false       | false        | false       | true         | true        |
+| Base.RefValue             | true        | false        | false       | false        | true        |

```

Comment:
there are a few mistakes, maybe you can check that using the function in julia given by
func = [Base.isconcretetype, Base.isprimitivetype, Base.isabstracttype, Base.isbitstype, Base.ismutabletype]